pkg/bbgo/interact.go

```diff
@@ -43,6 +61,25 @@ func NewCoreInteraction(environment *Environment, trader *Trader) *CoreInteracti
 	}
 }
 
+func (it *CoreInteraction) AddSupportedStrategyButtons(reply *interact.Reply, checkInterface interface{}) bool {
+	found := false
+	rt := reflect.TypeOf(checkInterface).Elem()
+	for signature, strategy := range it.exchangeStrategies {
+		if ok := reflect.TypeOf(strategy).Implements(rt); ok {
+			(*reply).AddButton(signature, "strategy", signature)
```

Comment:
can you pull out the interaction related code to the caller?

right now you need to pass a lot of object into this function, e.g. *reply.